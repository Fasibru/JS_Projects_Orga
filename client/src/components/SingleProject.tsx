import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

export interface SingleProjectProps {
  title: string;
  dateCreated: string;
}

const SingleProject: React.SFC<SingleProjectProps> = ({
  title,
  dateCreated,
}) => {
  const date = new Date(Number(dateCreated)).toLocaleDateString();
  return (
    <Fragment>
      <div>{date}</div>
      <div>{title}</div>
    </Fragment>
  );
};

SingleProject.propTypes = {
  title: PropTypes.string.isRequired,
  dateCreated: PropTypes.string.isRequired,
};

export default SingleProject;
