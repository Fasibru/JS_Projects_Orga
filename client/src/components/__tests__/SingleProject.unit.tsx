import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import SingleProject, { ProjectItem, ProjectLink } from '../SingleProject';

describe('Test <SingleProject />', () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  beforeEach(() => {
    wrapper = shallow(<SingleProject
      title="Test Title"
      dateCreated="1574614461353"
      dateModified="1574614461353"
    />);
  });

  it('Snapshot is consistent', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Project title is rendered correctly', () => {
    const projectTitle = wrapper.find(ProjectLink);

    expect(projectTitle.text()).toEqual('Test Title');
  });

  it('Project creation date is rendered correctly', () => {
    const projectDateCreated = wrapper.find(ProjectItem).at(0);

    expect(projectDateCreated.text()).toEqual('2019-11-24');
  });

  it('Project modification date is rendered correctly', () => {
    const projectDateModified = wrapper.find(ProjectItem).at(0);

    expect(projectDateModified.text()).toEqual('2019-11-24');
  });
});
