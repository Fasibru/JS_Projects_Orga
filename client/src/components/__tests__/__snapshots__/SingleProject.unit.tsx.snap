// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Headers /> Snapshot is consistent 1`] = `
<styled.div>
  <styled.a
    href=""
  >
    Test Title
  </styled.a>
  <styled.div>
    2019-11-24
  </styled.div>
  <styled.div>
    2019-11-24
  </styled.div>
</styled.div>
`;

exports[`Test <SingleProject /> Snapshot is consistent 1`] = `
<styled.div>
  <styled.a
    href=""
  >
    Test Title
  </styled.a>
  <styled.div>
    2019-11-24
  </styled.div>
  <styled.div>
    2019-11-24
  </styled.div>
</styled.div>
`;
